internal/upgrader/upgrader.go

```diff
@@ -54,31 +56,63 @@ func Upgrade(api CFClient, log Logger, cfg UpgradeConfig) error {
 		return err
 	}
 
-	if cfg.CheckDeactivatedPlans {
-		if err := checkDeactivatedPlans(log, serviceInstances); err != nil {
-			return err
-		}
-	}
-
-	totalServiceInstances := len(serviceInstances)
 	upgradableInstances := discoverInstancesWithPendingUpgrade(log, serviceInstances)
 
 	switch {
+	case cfg.CheckDeactivatedPlans:
+		return checkDeactivatedPlans(log, serviceInstances)
+	case cfg.FailIfNotUpToDate:
+		log.InitialTotals(len(serviceInstances), len(upgradableInstances))
+		defer log.FinalTotals()
+		var errs MultiError
+		errs.Append(performCheckUpToDate(upgradableInstances, log))
+		errs.Append(checkDeactivatedPlans(log, serviceInstances))
+		if len(errs.Errors) > 0 {
+			return &errs
+		}
+
+		return nil
+	case cfg.MinVersionRequired != "":
+		instances, err := filterInstancesVersionLessThanMinimumVersionRequired(serviceInstances, cfg.MinVersionRequired)
+		if err != nil {
+			return err
+		}
+		return performCheckUpToDate(instances, log)
```

Comment:
Doesn't actually check that anything is up to date, so change to a more meaningful name.

Reply:
@blgm Thank you very much for reviewing the code.

After analyzing the alternatives, I preferred removing the method instead of changing the name of a dummy method. It turned out that the code is now more readable, but I had to repeat the `dryRun` method call in the switch cases `case cfg.MinVersionRequired != ""` and `cfg.FailIfNotUpToDate`. In my opinion, this will help us in understanding the logic and in the next refactor, where we could improve the `dryRun` functionality. The logging strategy seems inconsistent. For example, in performDryRun, it logs upgrade failures for each service instance, which might be misleading as it's just a dry run.